src/transports/sentry_transport_winhttp.c

```diff
@@ -110,6 +111,21 @@ static int
 sentry__winhttp_transport_shutdown(uint64_t timeout, void *transport_state)
 {
     sentry_bgworker_t *bgworker = (sentry_bgworker_t *)transport_state;
+    winhttp_bgworker_state_t *state = sentry__bgworker_get_state(bgworker);
+
+    if (state->connect) {
+        WinHttpCloseHandle(state->connect);
+        state->connect = NULL;
+    }
+    if (state->session) {
+        WinHttpCloseHandle(state->session);
+        state->session = NULL;
+    }
+    if (state->request) {
+        WinHttpCloseHandle(state->request);
+        state->request = 0;
```

Comment:
Just a question: Shouldn’t these be closed in a particular order, in this case request first? Also, while it does not really make a difference, why is this reset to `0` while the others use `NULL`?

Also, since this can race with the background thread, might it be a good idea to exchange `state->request` first before closing it? So the racing background thread does not try to double-close it?

Reply:
Yes the order is important. Closing the session first will cancel all other requests in the queue as well.
When the request handle is closed, the background blocking function is unblocked and the loop continues. But because of no session handle all WinHttp requests will fail and the queue will get cleaned up.
This is not a nice solution, but still the best I could com up with.